thresholdkey_generator_test.go

```diff
@@ -11,44 +12,93 @@ var MockGenerateSafePrimes = func() (*big.Int, *big.Int, error) {
 	return big.NewInt(887), big.NewInt(443), nil
 }
 
+func TestCreateThresholdKeyGenerator(t *testing.T) {
+	var tests = map[string]struct {
+		keyLength     int
+		expectedError error
+	}{
+		"generator successfully created for 20 bit key length": {
+			keyLength: 20,
+		},
+		"generator can't be created for 19 bit key length": {
+			keyLength:     19,
+			expectedError: errors.New("Public key bit length must be an even number"),
+		},
+		"generator successfully created for 18 bit key length": {
+			keyLength: 18,
+		},
+		"generator can't be created for 17 bit key length": {
+			keyLength:     17,
+			expectedError: errors.New("Public key bit length must be an even number"),
+		},
+		"generator can't be created for 16 bit key length": {
+			keyLength:     16,
+			expectedError: errors.New("Public key bit length must be at least 18 bits"),
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
```

Comment:
We should also add a positive case validation that the generator was created with all passed values.

Reply:
Yes we have, but we're checking only that `gen != nil`.
We're not checking if the values passed to `GetThresholdKeyGenerator(test.keyLength, 4, 3, rand.Reader)` are set on the `gen`, e.g.
```go
if gen.publicKeyBitLength != test.keyLength {
    t.Fatalf(....)
}
```
etc.